### PR TITLE
Merge xslt/ into dom/xslt/

### PR DESCRIPTION
**Pull request by foolip**
After https://github.com/web-platform-tests/wpt/pull/28638 the interface
is in dom.idl, so remove xslt.tentative.idl and move the remaining tests
into dom/xslt/ with a new README.

### CODE REVIEW

**Comment by foolip** [comment]
@annevk https://github.com/web-platform-tests/wpt/pull/28638 stands on its own and doesn't break tests so this can be landed separately, but I'd still like to land them in quick succession.

File: xslt/META.yml [hunk 1]
```diff
@@ -1,4 +1 @@
-# NOTE: There is no standard for this API, only:
-# https://wiki.whatwg.org/wiki/DOM_XSLTProcessor
-# and the notes in the HTML5 spec, linked.
-spec: https://html.spec.whatwg.org/multipage/scripting.html#scriptTagXSLT
```
Comment (annevk) [inline xslt/META.yml:4]: This is still a good reference as well as it defines some things around `script`, no?

**Review by annevk — APPROVED** [review]
How about we move xslt/ to dom/xslt?

**Comment by foolip** [comment]
> How about we move xslt/ to dom/xslt?

I considered it, there are certainly few enough tests that it would be fine to do so, but the precedent of domxpath/ and the fact that https://www.w3.org/TR/xslt/ exists made me not. But hey, let's.

**Review by annevk — APPROVED** [review]

**Pull request merged**